### PR TITLE
Fix memory leak in GpuIntervalUtilsTest

### LINKED ISSUE #7603: [BUG] GpuIntervalUtilsTest leaks memory
**Describe the bug**
```
GpuIntervalUtilsTest:
- testToDayTimeIntervalString for from second to second
Run completed in 4 seconds, 487 milliseconds.
Total number of tests run: 1
Suites: completed 1, aborted 0
Tests: succeeded 1, failed 0, canceled 0, ignored 0, pending 0
All tests passed.
23/01/27 11:53:10.128 Thread-2 ERROR DeviceMemoryBuffer: A DEVICE BUFFER WAS LEAKED (ID: 3 7fd35f600000)
23/01/27 11:53:10.134 Thread-2 ERROR MemoryCleaner: Leaked device buffer (ID: 3): 2023-01-27 11:53:09.0838 PST: INC
java.lang.Thread.getStackTrace(Thread.java:1564)
ai.rapids.cudf.MemoryCleaner$RefCountDebugItem.<init>(MemoryCleaner.java:333)
ai.rapids.cudf.MemoryCleaner$Cleaner.addRef(MemoryCleaner.java:91)
ai.rapids.cudf.MemoryBuffer.incRefCount(MemoryBuffer.java:275)
ai.rapids.cudf.MemoryBuffer.<init>(MemoryBuffer.java:117)
ai.rapids.cudf.BaseDeviceMemoryBuffer.<init>(BaseDeviceMemoryBuffer.java:30)
ai.rapids.cudf.DeviceMemoryBuffer.<init>(DeviceMemoryBuffer.java:120)
ai.rapids.cudf.Rmm.alloc(Rmm.java:326)
ai.rapids.cudf.DeviceMemoryBuffer.allocate(DeviceMemoryBuffer.java:143)
ai.rapids.cudf.DeviceMemoryBuffer.allocate(DeviceMemoryBuffer.java:133)
ai.rapids.cudf.HostColumnVector.copyToDevice(HostColumnVector.java:197)
ai.rapids.cudf.HostColumnVector$Builder.buildAndPutOnDevice(HostColumnVector.java:1997)
ai.rapids.cudf.ColumnVector.build(ColumnVector.java:1115)
ai.rapids.cudf.ColumnVector.fromLongs(ColumnVector.java:1240)
com.nvidia.spark.rapids.GpuIntervalUtilsTest.$anonfun$testDayTimeToString$1(GpuIntervalUtilsTest.scala:37)
```

**Steps/Code to reproduce bug**

```bash
mvn install -DskipTests -Dbuildver=331
mvn test -pl tests  -Dbuildver=331 -Dsuites='com.nvidia.spark.rapids.GpuIntervalUtilsTest @ testToDayTimeIntervalString for from second to second'
```

**Expected behavior**
should not exhibit mem leak warnings in the log

**Environment details (please complete the following information)**
 local dev

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by gerashegalov**
Fixes #7603.

don't leak `micros` backing buffers 

Signed-off-by: Gera Shegalov <gera@apache.org>

### CODE REVIEW

File: tests/src/test/330+/scala/com/nvidia/spark/rapids/GpuIntervalUtilsTest.scala [hunk 1]
```diff
@@ -34,17 +34,19 @@ class GpuIntervalUtilsTest extends FunSuite with Arm {
     val caseName = s"testToDayTimeIntervalString for " +
```
Comment (jlowe) [inline tests/src/test/330+/scala/com/nvidia/spark/rapids/GpuIntervalUtilsTest.scala:34]: 2023 copyrights

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Comment by gerashegalov** [comment]
build

**Comment by gerashegalov** [comment]
build

**Pull request merged**